dmscripts/notify_buyers_to_award_closed_briefs.py

```diff
@@ -0,0 +1,139 @@
+
+        for user in brief_users:
+            if user['active']:
+                try:
+                    notify_client.send_email(
+                        user['emailAddress'], notify_template_id, context_data, allow_resend=False
+                    )
+                except EmailError:
+                    failed_users.append(user['id'])
+
+    return failed_users
+
+
+def _check_that_brief_closing_date_before_the_offset(date_closed, offset_days):
+    if date_closed:
+        date_closed = date_helpers.get_date_closed(date_closed)
+        if date_closed > (datetime.utcnow() - timedelta(days=offset_days)).date():
+            return False
+        return date_closed
+    return date.today() - timedelta(days=offset_days)
+
+
+def main(
+    data_api_url, data_api_access_token, notify_api_key, notify_template_id, date_closed=None, dry_run=None,
+    user_id_list=None, offset_days=28
+):
+    """
+    Send emails to buyers via Notify, reminding them to award their closed briefs
+    """
+    logger.info("Data API URL: {data_api_url}", extra={'data_api_url': data_api_url})
+
+    date_closed = _check_that_brief_closing_date_before_the_offset(date_closed, offset_days)
+    if not date_closed:
+        logger.error('Not allowed to notify about briefs that closed less than {} days ago', offset_days)
+        return False
+
+    data_api_client = dmapiclient.DataAPIClient(data_api_url, data_api_access_token)
+    if not dry_run:
+        notify_client = DMNotifyClient(notify_api_key, logger=logger)
+
+    closed_briefs = brief_data_helpers.get_closed_briefs(data_api_client, date_closed)
+    if not closed_briefs:
+        logger.info("No briefs closed on {date_closed}", extra={"date_closed": date_closed})
+        return True
+
+    # If user list supplied, check that the briefs have at least one user in the list
+    closed_briefs = _filter_briefs_by_optional_user_id_list(closed_briefs, user_id_list)
+
+    logger.info("Notifying users about {briefs_count} closed brief(s)", extra={'briefs_count': len(closed_briefs)})
+
+    failed_users_for_each_brief = {}
+    all_failed_users = []
+    for brief in closed_briefs:
+        logging_context = {
+            'brief_title': brief['title'],
+            'brief_id': brief['id'],
+            'no_of_users': len(brief['users']),
+        }
+        if dry_run:
+            logger.info(
+                "Would notify {no_of_users} user(s) about brief ID: {brief_id} - '{brief_title}'",
+                extra=logging_context
+            )
+        else:
+            logger.info(
+                "Notifying {no_of_users} user(s) about brief ID: {brief_id} - '{brief_title}'",
+                extra=logging_context
+            )
+            failed_users = notify_users(notify_client, notify_template_id, brief, user_id_list)
+            if failed_users:
+                failed_users_for_each_brief[brief['id']] = failed_users
+                all_failed_users.extend(failed_users)
```

Comment:
Wonder if we need both of these?
Are you expecting people to use information from both of the two error formats?

Reply:
Yeah, I wasn't sure about this - depends what's more likely to be useful. A full list of users who failed so we can c+p it straight into the script to re-run, or do we want to know the individual briefs as well (so we can figure out why those users failed)?

Reply:
Fair point.

Probably worth just putting this out there and then seeing how our script fairs and if we ever get any errors, checking we can then recover from them. I think the fact that notify won't let you send the same email twice to a user may be helpful (I believe)... meaning in most cases we could likely just run the whole script again if we get a random failure from notify (as opposed to a persistent one such as bad data).